api/types.proto

```diff
@@ -465,12 +465,31 @@ enum TaskState {
 	// Remove this message when the states are deemed perfect.
 }
 
+// State of Networks attached to a container
+message ContainerNetworkStatus {
```

Comment:
I have literally no idea why I chose to flatten this in the way I did, I probably started down one path before I had fully groked the CNI data structures and never reevaluated the base decision.

I'll do as you suggest and unflatten it making it a (non-`oneof`) `string error`  and `repeated string ip`.